File: app/src/main/java/net/osdn/gokigen/blecontrol/lib/ble/connect/ICameraPowerOn.java
package net.osdn.gokigen.blecontrol.lib.ble.connect;

public interface ICameraPowerOn
{
    // カメラ起動指示
    void wakeup(PowerOnCameraCallback callback);

    // 実行終了時のコールバックのインタフェース
    interface PowerOnCameraCallback
    {
        void wakeupExecuted(boolean isExecute);
    }
}

File: app/src/main/java/net/osdn/gokigen/blecontrol/lib/ble/connect/ICameraPowerOn.kt
package net.osdn.gokigen.blecontrol.lib.ble.connect

interface ICameraPowerOn {
    // カメラ起動指示
    fun wakeup(callback: PowerOnCameraCallback)

    // 実行終了時のコールバックのインタフェース
    interface PowerOnCameraCallback {
        fun wakeupExecuted(isExecute: Boolean)
    }
}
